research/06-require-ana.tex

```diff
@@ -1,9 +1,184 @@
+state to be saved and restored.
+
+\subsubsection{Could have}
+
+Requirements listed under ``could-haves'' are desirable, but not necessary.
+These requirements often improve usability or customer satisfaction and are
+included only if time permits.
+
+\paragraph{Context-sensitive code completion} The multiline input editor should
+ideally function just like an IDE's editor. Context-sensitive code completion is
+therefore a nice feature to offer the users. Context-sensitive code-completion
+also helps with the discoverability of APIs and therefore is another step
+towards the explorative and interactive environment a REPL is supposed to
+provide.
+
+\todo{Quite the same as environment inspection maybe?}
+\paragraph{Hover over variables to see value, type and others} Another step
+towards debugging would be the ability to hover variables with the mouse in
+order to inspect their value, type and other known information. The difference
+between this feature and the previously mentioned environment inspection is that
+this feature works per variable.
+
+\paragraph{Literate programming} As explained in section
+\cref{sec:literate-programming}, literate programming offers the advantage that
+code and documentation go hand in hand. This allows developers of languages in
+Spoofax to document and illustrate their language simultaneously with the
+development: documentation and examples can never be outdated, because outdated
+example code will halt the execution.
+
+\paragraph{Integration with other IDEs (IntelliJ)} As mentioned before, Spoofax
+is currently undergoing a transition not to be tied to a specific IDE.
+Generation of Spoofax's editor services to IntelliJ are currently a work in
+progress and it would be nice if the generated REPLs work in IntelliJ from the
+start.
+
+\subsubsection{Won't have}
+
+Requirements listed under ``won't-haves'' are not for implementation in this
+project. They have been identified as possible features, but are outside of the
+scope of this project and listed only as possible suggestions for further work.
+
+\paragraph{GDB-style debugging and nested REPLs} Spoofax currently does not
+generate any debugging features, which would be required for REPLs to offer such
+functionality as a built-in feature. Writing such debugging generators is
+outside the scope of this project and as such, offering debugging capabilities
+inside the REPLs is something to consider for a later version.
 
 \subsection{Minimal viable product}
 \label{ssec:mvp}
 
+The design goals and requirements listed in the previous subsections give a good
+idea of what the deliverable should look like. It is possible, however, that due
+to unforeseen problems, not all the listed requirements and design goals can be
+met. It is important to therefore define a minimal subset of the design goals
+and requirements that \emph{must} be present in the final deliverable: the ``must
+have'' requirements have to be implemented, as do the first two design goals.
```

Comment:
design goals aren't really something you 'implement', they are rather something you adhere to.

I'm not sure how to reformulate correctly.